tools/gulptasks/dist-compress.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) Highsoft AS
+ */
+
+const gulp = require('gulp');
+const fs = require('fs-extra');
+const zlib = require('zlib');
+const glob = require('glob');
+const log = require('./lib/log');
+const build = require('./lib/build');
+const yargs = require('yargs');
+const zip = require('gulp-zip');
+const stream = require('stream');
+const util = require('util');
+
+
+const DIST_DIR = 'build/dist';
+
+
+/**
+ * Creates zip files for highcharts, highstock, highmaps, highcharts-gantt.
+ *
+ * @return {Promise<any> | Promise | Promise} Promise to keep
+ */
+function distZip() {
+    const properties = build.getBuildProperties();
+
+    const zipTasks = Object.keys(properties).map(key => {
+
+        const { product } = properties[key];
+        const { name, version } = product;
+
+        const dirToZip = `${DIST_DIR}/${key}`;
+        if (!fs.existsSync(dirToZip)) {
+            return Promise.reject(new Error(`Missing folder: ${dirToZip}. Has the other dist tasks been run in advance?`));
+        }
+
+        const zipFileName = `${name.replace(/ /g, '-')}-${version}.zip`;
+        const excludedDir = `!${dirToZip}/js-gzip/**`;
+        log.message(`Zipping file: ${zipFileName}, excluding sub-directory ${excludedDir}`);
+
+        return new Promise((resolve, reject) => {
+            gulp.src([dirToZip + '/**', excludedDir])
+                .pipe(zip(zipFileName))
+                .pipe(gulp.dest(DIST_DIR))
+                .on('error', reject)
+                .on('end', resolve);
+        });
+    });
+
+    return Promise.all(zipTasks);
```

Comment:
If you add here a `.then(() => undefined);`, you can change the return type to `Promise<void>`.